stackslib/src/chainstate/stacks/boot/pox-4.clar

```diff
@@ -583,7 +606,10 @@
       (try! (can-stack-stx pox-addr amount-ustx first-reward-cycle lock-period))
 
       ;; ensure the signer key can be used
-      (try! (insert-signer-key signer-key))
+      (try! (insert-signer-key signer-key first-reward-cycle))
+
+      ;; update the total ustx for the signer key
+      (increment-signer-key-total-ustx first-reward-cycle signer-key amount-ustx)
```

Comment:
You're going to have to do this for all of the reward cycles in which the user has stacked